### PR TITLE
Return nil on empty get_area()

### PR DESCRIPTION
**Pull request by everamzah**

### CODE REVIEW

**Comment by everamzah** [comment]
Maybe fix https://github.com/minetest/minetest/issues/4507

File: src/script/lua_api/l_areastore.cpp [hunk 1]
```diff
@@ -110,6 +110,9 @@ int LuaAreaStore::l_get_area(lua_State *L)
 
 	const Area *res;
 
+	if (!ast->getArea(id))
+		return 0;
+
 	res = ast->getArea(id);
```
Comment (rubenwardy) [inline src/script/lua_api/l_areastore.cpp]: it would be better to add the check after here, and only call getArea once

Comment (everamzah) [inline src/script/lua_api/l_areastore.cpp]: Okay, thank you.  I will try and wrap my head around that!

Comment (rubenwardy) [inline src/script/lua_api/l_areastore.cpp]: should be like

```lua
res = ast->getArea(id);
if (!res)
	return 0;
```

Comment (everamzah) [inline src/script/lua_api/l_areastore.cpp]: Well I force pushed that change, anyway, though I'm still compiling on this old laptop to test.

**Comment by Zeno-** [comment]
lgtm :+1:

**Comment by kwolekr** [comment]
Could you add "Returns nil if specified area id does not exist." to the documentation, please?
Other than that, LGTM.

**Comment by everamzah** [comment]
Okie doke, lua_api.txt updated.

**Pull request merged**